### PR TITLE
Added test from Ben Brelje

### LINKED ISSUE #1340: src_indices notation allows non-local transfers to be defined on local groups
### Summary of Issue

When running in parallel, a non distributed component is allowed to specify src_indices in excess of the size of the incoming vector. For a distributed component, this is supposed to specify a transfer from another processor. For a non-distributed component, this undefined behavior can create erroneous results.

### Issue Type

- [ X] Bug

### Description

A group with all non-distributed components and no parallel groups will use the default_vector class for variable transfers. However, when run in parallel, the non-distributed input variables are not prevented from specifying src_indices in excess of the size of the incoming variable connection. 

This results in the default value of the variable being read for all time even if the variable value is updated on the other processor rank (since there is no Petsctransfer to overwrite that portion of the local vector with the correct value)

### Example

See [gist](https://gist.github.com/bbrelje/01ed972b8ac0b4870a9983dc068c4fa8)

Note that if SRC_INDICES is set to 0 the value on each proc is returned (default)
Alternatively, if distributed=True, src_indices works as intended for distributed systems and returns the result on proc=SRC_INDICES for all procs.

### Proposed Fix

Do not allow src_indices specification out of bounds of the variable input shape, unless the component is distributed and the PETSc transfer exists for the owning group. Raise an error on setup?

### Environment

Operating System: Ubuntu 16.04

Python environment: Anaconda Python 3.7

Packages: OpenMDAO 9931ddd4b246485d651930a91ffde3dcec1bb07c

### PR DESCRIPTION
**Pull request by DKilkenny**
### Summary

Added test from Ben Brelje to capture error message when running in parallel with non distributed components

### Related Issues

- Resolves #1340

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/core/tests/test_connections.py [hunk 1]
```diff
@@ -613,5 +613,66 @@ def compute_partials(self, inputs, J):
...
+        if MPI.COMM_WORLD.rank == 0:
+            setval = np.array([2.0,3.0])
+        else:
+            setval = np.array([10.0,20.0])
+
+        # no parallel or distributed comps, so default_vector is used (local xfer only)
+        model.add_subsystem('p1', om.IndepVarComp('x', setval))
+        model.add_subsystem('c3', TestComp())
+        if FORCE_PETSCTRANSFER:
+            model.add_subsystem('c4', TestCompDist())
```
Comment (swryan) [inline openmdao/core/tests/test_connections.py:666]: I think the intent was to add a test that runs 
```under MPI (without adding any distributed comps or parallelgroups to the model) and see if a correct error is raised.```
(quoting @Kenneth-T-Moore from #1467 discussion)
So the `FORCE_PETSCTRANSFER=False` version of @bbrelje 's gist?

Comment (swryan) [reply openmdao/core/tests/test_connections.py:666]: hmm I see the `COMP_DISTRIBUTED` flag does this.. so not clear on the function of the FORCE_PETSCTRANSFER flag

Comment (DKilkenny) [reply openmdao/core/tests/test_connections.py:666]: Yeah I could clean that up a bit. I'll fix it

**Comment by bbrelje** [comment]
@swryan @DKilkenny This looks correct to me, sorry for the delay in the review.

**Pull request merged**